### PR TITLE
prep to release the latest test_api

### PR DESCRIPTION
**Pull request by jakemac53**
Also release a new test_core and test that depend on it.

Fixes an issue when `Error`s are serialized as remote exceptions. We now allow an empty message and properly de-serialize it.

### CODE REVIEW

**Comment by jakemac53** [comment]
I am going to wait until tomorrow AM to land and publish

**Comment by jakemac53** [comment]
Actually I am going to wait for https://github.com/dart-lang/test/pull/1454 and roll that together with this as well

**Review by natebosch — APPROVED** [review]

**Pull request merged**
